scripts/entity/components/movement_components/player_movement_component.gd

```diff
@@ -0,0 +1,62 @@
+class_name PlayerMovementComponent extends BaseMovementComponent
+## The movement implementation for a human player.
+
+## The max horizontal speed the entity can move.
+@export var MAX_SPEED: int = 200
+## The jump velocity of the entity.
+@export var JUMP_VELOCITY: int = -300
+## The horizontal movement acceleration for the entity.
+@export var ACCELERATION: int = 1500
+## The time it takes to fully decelerate from horizontal input in seconds.
+@export_range(0, 10, 0.1, "or_greater") var DECELERATION_TIME: float = 0.2
+
+## The current input direction.
+var current_input_direction: float = 0
+## How fast the character decelerates after loss of input.
+var deceleration: float = MAX_SPEED / DECELERATION_TIME
+
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	#do nothing
+	pass
+
+## Applies movement to the character.
+func _physics_process(delta: float) -> void:
+	super._physics_process(delta)
+	apply_gravity(delta)	
+	apply_horizontal_movement(delta)
+	move_and_slide()
+
+func on_movement_input(horizontal_vector2: float) -> void:
+	if(current_input_direction == 0 && horizontal_vector2 != 0):
+		on_horizontal_movement_started.emit(horizontal_vector2)
+	elif(current_input_direction != 0 && horizontal_vector2 == 0):
+		on_horizontal_movement_cancelled.emit(horizontal_vector2)
+	elif current_input_direction != 0 && horizontal_vector2 != 0:
+		on_horizontal_movement.emit(horizontal_vector2)
```

Comment:
Maybe I'm misreading these but these feel mixed up to me.

If inputDir == 0 (so not trying to move) and vector != 0 (so was moving/are moving) we do horizontal movement started?

The 3rd if seems right, but the first/second seem mixed up unless I'm misunderstanding.

Reply:
I tested it and it looks right to me

Reply:
The on move one looks incorrect though

Reply:
Moved it since the functionality was supposed to be based on horizontal movement